### PR TITLE
no_log even when task_result doesn't provide key

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
 - now also checks task property
 - added reproducer to tests for unreachable status on item loop

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

task result
##### ANSIBLE VERSION

```
>=2.4
```

### CODE REVIEW

File: test/integration/targets/no_log/no_log_local.yml [hunk 1]
```diff
@@ -63,3 +63,30 @@
...
+            login: three
+            password: DO_NOT_LOG_UNREACHABLE_ITEM
+      lineinfile:
+        path: /dev/null
+        mode: 0600
+        create: true
+        insertafter: EOF
+        line: "machine {{ item.machine }} login {{ item.login }} password {{ item.password }}"
+      loop: "{{ logins }}"
+      no_log: true
```
Comment (dmsimard) [inline test/integration/targets/no_log/no_log_local.yml:92]: Should there be an assert that checks that ``DO_NOT_LOG_UNREACHABLE_ITEM`` is not in the output of the task somehow ?

Comment (bcoca) [reply test/integration/targets/no_log/no_log_local.yml:92]: in runme.sh, it was already part of existing tests

Comment (dmsimard) [reply test/integration/targets/no_log/no_log_local.yml:92]: ah, gotcha.

**Comment by abadger** [comment]
+1 to merge

**Comment by abadger** [comment]
Also needs backporting to stable-2.6, stable-2.5 and stable-2.4

**Review by achinthagunasekara — APPROVED** [review]

**Comment by abadger** [comment]
Hmm,.. this does need a change log fragment, though.  (Which will make backporting to 2.6 and 2.5 easier but will have to be changed for 2.4)

**Comment by abadger** [comment]
The unstable test result is unrelated merging and openin backports.

**Pull request merged**